src/stan/model/indexing/lvalue.hpp

```diff
@@ -383,10 +386,12 @@ template <
 inline void assign(Mat1&& x,
                    const cons_index_list<index_max, nil_index_list>& idxs,
                    const Mat2& y, const char* name = "ANON", int depth = 0) {
-  stan::math::check_range("matrix[max] assign range", name, x.cols(),
+  stan::math::check_range("matrix[max] assign row", name, x.rows(),
```

Comment:
cols -> rows, was this a bug?

Reply:
Yeah the issue is that we use matrices of the same dimensions in the test so it couldn't tell. I'll change those so this get's caught